sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -549,15 +549,18 @@ class HDFSFileCatalog(
 
   refresh()
 
+  private def isInvalidFile(f: FileStatus): Boolean =
+    f.getPath.getName.startsWith("_") || f.getLen == 0
```

Comment:
The new FileSourceStratyge expects non-empty files [here](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala#L141)